packages/chord-symbol/src/typedefs.js

```diff
@@ -1,13 +1,26 @@
 /**
- * A data object representing a chord. It is the result of the parsing operation and can be used for rendering.
+ * An object that may be a chord data object or a chord parsing failure object. The `success` property indicates which.
+ * @typedef {Chord | ChordParseFailure} MaybeChord
+ */
+
+/**
+ * A data object representing a chord. It is the result of a successful parsing operation and can be used for rendering.
  * @typedef {Object} Chord
  * @type {Object}
+ * @property {true} [success] - indicates that the parsing was successful.
```

Comment:
Can you explain why a new `success` property is needed and why we cannot rely on the `error` property instead? tbh I'm a bit reluctant to introduce new code just for typescript. And btw I don't see where this property is actually set in the chord object?

Reply:
My apologies. My head hasn't been screwed on properly recently. 🙄
Yes, I forgot to actually ever set the `success` property.
And no, it isn't actually needed. I had forgotten to make the `error` property required instead of optional, so it couldn't discriminate on it and I added the `success` property. I will remove it 👍